File: eva-web/eva-web-auth/src/main/java/io/nerv/exception/OathException.java
package io.nerv.exception;

import io.nerv.core.enums.BizCode;
import lombok.Getter;
import org.springframework.security.core.AuthenticationException;

/**
 * 自定义鉴权异常类
 */
@Getter
public class OathException extends AuthenticationException {
    // 错误码
    private String code;
    // 消息内容
    private String msg;

    public OathException(String msg) {
        super(msg);
    }

    public OathException(BizCode errorCodeEnum) {
        super(errorCodeEnum.getName());
        this.code = errorCodeEnum.getIndex();
        this.msg = errorCodeEnum.getName();
    }
}

File: eva-core/src/main/kotlin/io/nerv/exception/OathException.kt
package io.nerv.exception

import io.nerv.core.enums.BizCode
import lombok.Getter
import org.springframework.security.core.AuthenticationException

/**
 * 自定义鉴权异常类
 */
@Getter
class OathException : AuthenticationException {
    // 错误码
    var code: String? = null

    // 消息内容
    var msg: String? = null

    constructor(msg: String?) : super(msg) {}

    constructor(errorCodeEnum: BizCode) : super(errorCodeEnum.getName()) {
        code = errorCodeEnum.getIndex()
        msg = errorCodeEnum.getName()
    }
}